components/automate-gateway/gateway/middleware/auth_interceptor.go

```diff
@@ -37,60 +36,6 @@ func NewAuthInterceptor(
 	return &authInterceptor{authn: authn, authz: authz}
 }
 
-type SwitchingAuthorizationHandler interface {
```

Comment:
The unnecessary types are removed, the rest is moved to the already-present `types.go` in the same directory.